skbeam/core/accumulators/bin_image.py

```diff
@@ -0,0 +1,219 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+
+
+class BinImage(object):
+    """
+    Class to bin an ndarray onto an axis using a
+    user-specified bin for each element of the array.
+    """
+
+    def __init__(self, bin_values, norm, nbins, min=None, max=None,
+                 weights=None):
+        """
+        Parameters
+        ----------
+        bin_values : np.ndarray
+            For each pixel, this is the value that determines which bin
+            a pixel's intensity will contribute to.  It can be an arbitrary
+            array of floats/ints which will be discretized into nbins equally
+            sized bins.
+        norm : bool
+            If set to true, normalize histogram by the sum of the weights
+            of all pixels contributing to a bin (see "weights" parameter).
+        nbins : int
+            The number of histogram bins.
+        min/max : int/float, optional
+            Lower/Upper limits of returned histogram.  If not specified
+            one will be computed from bin_values/min/max/weights
+        weights : np.ndarray, optional
+            A weight for each pixel with the same shape as bin_values.
+            These can be set to zero to ignore a pixel, similar to a mask.
+            If not provided, each pixel is assigned a weight of 1.
+        """
+
+        self._bin_values = bin_values
+        self._weights = weights
+        self._nbins = nbins
+        self._norm = None
+        self._min = min
+        self._max = max
```

Comment:
You might want to rename your min and max variables, they shadow the existing python min and max functions